### PR TITLE
[F] Updated queries to remove references to removed pool attributes (ENT-1837)

### PR DESCRIPTION
**Pull request by Ceiu**
- Updated the entitlement curator and pool curator to no longer
  reference the cp_pool.provided_products column, or the
  cp2_pool_provided_products or cp2_pool_derprov_products tables

### CODE REVIEW

**Comment by Ceiu** [comment]
This PR is currently blocked by #2890. Once that PR is merged, this PR should be marked "ready for review" and merged if appropriate.

**Review by Januson — APPROVED** [review]

**Comment by nikosmoum** [comment]
Just a note on this: The org_migration.py tool still references these tables. we should eventually update it

**Pull request merged**
